src/benchmarks/micro/libraries/System.Memory/SpanHelpers.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+
+namespace System.Memory
+{
+    [GenericTypeArguments(typeof(byte))]
+    [GenericTypeArguments(typeof(char))]
+    [BenchmarkCategory(Categories.Runtime, Categories.Libraries, Categories.Span)]
+    [ShortRunJob]
+    public unsafe class SpanHelpers<T>
+        where T : unmanaged, IComparable<T>, IEquatable<T>
+    {
+        private T* _searchSpace;
+        private T _value;
+
+        [ParamsSource(nameof(LengthValues))]
+        public int Length { get; set; }
+
+        public static IEnumerable<object> LengthValues()
+        {
+            // The values for the length take into account the different cut-offs
+            // in the vectorized pathes.
+
+            if (typeof(T) == typeof(byte))
+            {
+                // Vectorization is done on 2 * Vector<byte>.Count => 64 byte elements
+                yield return 63;    // one less the vectorization threshould
```

Comment:
At the moment these values are based on `Vector<T>.Count` on x64 with AVX2 enabled.
So for e.g. ARM where `Vector<byte>.Count` is 16 (instead of 32 as on x64) the values should be halfed.

Via `ParamsSource` this could be taken into account. Shall I update to this?
(Didn't do now, because of the discussion in the PR's description -- i.e. I'd like to an outcome from this first).

Reply:
@adamsitnik might suggest.

Reply:
@adamsitnik could you advise, so we can get this merged?